lib/zipkin-tracer/trace_wrapper.rb

```diff
@@ -0,0 +1,18 @@
+module ZipkinTracer
+  class TraceWrapper
+    def self.wrap_in_custom_span(config, span_name, span_kind: Trace::Span::Kind::SERVER, app: nil)
+      raise ArgumentError, "you must provide a block" unless block_given?
+
+      zipkin_config = ZipkinTracer::Config.new(app, config).freeze
+      tracer = ZipkinTracer::TracerFactory.new.tracer(zipkin_config)
+      trace_id = ZipkinTracer::TraceGenerator.new.next_trace_id
+
+      ZipkinTracer::TraceContainer.with_trace_id(trace_id) do
+        tracer.with_new_span(trace_id, span_name) do |span|
```

Comment:
We need to check `sampled?` here.
Also, some might want to add additional information like in [local-tracing](https://github.com/openzipkin/zipkin-ruby#local-tracing).
How about passing `span` to the `yield`?

```ruby
if trace_id.sampled?
  tracer.with_new_span(trace_id, span_name) do |span|
    span.kind = span_kind
    yield(span)
  end
else
  yield(ZipkinTracer::NullSpan.new)
end
```

Reply:
Great suggestions! Implemented in 8598529a56039cb08be9147b5a9aca3f0d230fcd